astropy/io/fits/tests/test_connect.py

```diff
@@ -67,12 +66,19 @@ def setup_class(self):
         )
 
     def test_overwrite_with_path(self, tmp_path):
-        filename = "temp.fits"
+        filename1 = tmp_path / "temp1.fits"
```

Comment:
@neutrinoceros - IIRC #17552 was fixing the case of writing without a leading directory, i.e. `t1.write("foo.fits", format="fits")`. It seems you are removing this test case ?

Reply:
I didn't think this part was important, but since I also needed a rebase to fix a conflict, I updated the test as well so the second part (using `chdir`) is all about relative paths now.